flux-plugin/kubeflux/kubeflux.go

```diff
@@ -110,6 +110,13 @@ func (kf *KubeFlux) PreFilterExtensions() framework.PreFilterExtensions {
 }
 
 func (kf *KubeFlux) askFlux(ctx context.Context, pod *v1.Pod, filename string) (string, error) {
+	// clean up previous match if a pod has already allocated previously
+	if _, ok := kf.podNameToJobId[pod.Name]; ok {
+		fmt.Println("Clean up previous allocation")
+		kf.mutex.Lock()
```

Comment:
I have a question, I post it attached to this line but it's valid for the entire commit.
We do we need a lock? I guess I am missing some Kubernetes internals here.

Reply:
I added this lock because of the following reason: The informer runs inside it's own goroutine, and it will modify `podNameToJobId` map by the event handler. At the same time, main goroutine (the one calls `askFlux`) may also modify the `podNameToJobId` map. I think I should use `mutex` in the map read in line 114 according to [this post](https://stackoverflow.com/questions/11063473/map-with-concurrent-access).